agent/agent.go

```diff
@@ -389,6 +395,12 @@ const recycleTimeout = 10 * time.Minute
 // statusQueryReplyTimeout is the amount of time to wait for status query reply.
 const statusQueryReplyTimeout = 30 * time.Second
 
+// nodeTimeout specifies the amout of time to wait for a node status query reply.
```

Comment:
Nit: I would explain the intent in a comment to highlight that each of the status/node/probe timeouts is supposed to be smaller than the "previous" one to allow for collection to return result before deadline. Same below.